File: CS/WindowsApplication1/MyColorEdit/RepositoryItemMyColorEdit.cs
using System;
using DevExpress.XtraEditors.Registrator;
using DevExpress.XtraEditors.Repository;

namespace CustomEditors
{
    [UserRepositoryItem("Register")]
    public class RepositoryItemMyColorEdit : RepositoryItemColorEdit
    {
        static RepositoryItemMyColorEdit()
        {
            Register();
        }
        public RepositoryItemMyColorEdit() { }

        internal const string EditorName = "MyColorEdit";

        public static void Register()
        {
            EditorRegistrationInfo.Default.Editors.Add(new EditorClassInfo(EditorName, typeof(MyColorEdit),
                typeof(RepositoryItemMyColorEdit), typeof(DevExpress.XtraEditors.ViewInfo.ColorEditViewInfo),
                new DevExpress.XtraEditors.Drawing.ColorEditPainter(), true, null));
        }
        public override string EditorTypeName
        {
            get { return EditorName; }
        }

        protected override string GetColorDisplayText(System.Drawing.Color editValue)
        {
            return ColorLocalizer.GetLocalizedColorName(editValue);
        }
    }
}

File: VB/WindowsApplication1/MyColorEdit/RepositoryItemMyColorEdit.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.XtraEditors.Registrator
Imports DevExpress.XtraEditors.Repository

Namespace CustomEditors
	<UserRepositoryItem("Register")> _
	Public Class RepositoryItemMyColorEdit
		Inherits RepositoryItemColorEdit
		Shared Sub New()
			Register()
		End Sub
		Public Sub New()
		End Sub

		Friend Const EditorName As String = "MyColorEdit"

		Public Shared Sub Register()
            EditorRegistrationInfo.Default.Editors.Add(New EditorClassInfo(EditorName, GetType(MyColorEdit), GetType(RepositoryItemMyColorEdit), GetType(DevExpress.XtraEditors.ViewInfo.ColorEditViewInfo), New DevExpress.XtraEditors.Drawing.ColorEditPainter(), True))
		End Sub
		Public Overrides ReadOnly Property EditorTypeName() As String
			Get
				Return EditorName
			End Get
		End Property

		Protected Overrides Function GetColorDisplayText(ByVal editValue As System.Drawing.Color) As String
			Return ColorLocalizer.GetLocalizedColorName(editValue)
		End Function
	End Class
End Namespace
